### PR TITLE
Updates to latest maven versions aligned with latest docker images

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This also updates example docker images to latest

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
raised https://github.com/tcort/markdown-link-check/issues/377 for the false negative on lint

**Comment by codefromthecrypt** [comment]
@reta @making @shakuzen  what's left is the UI (zipkin-lens) CVE which cannot be automatically fixed as it seems there's a breaking change. This PR is against the main repo, so PRs can go against it if someone doesn't have access to add a commit. Once done, back on track. Appreciate a hand pinging someone to help with this, or doing it directly.

```
┌───────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬─────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability │ Severity │ Status │ Installed Version │ Fixed Version │ Title │
├───────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼─────────────────────────────────────────────────────────────┤
│ cross-spawn │ https://github.com/advisories/GHSA-3xgq-45jj-v275 │ HIGH │ fixed │ 6.0.5 │ 7.0.5, 6.0.6 │ Regular Expression Denial of Service (ReDoS) in cross-spawn │
│ │ │ │ │ │ │ https://avd.aquasec.com/nvd/cve-2024-21538 │
│ │ │ │ ├───────────────────┤ │ │
│ │ │ │ │ 7.0.3 │ │ │
│ │ │ │ │ │ │ │
├───────────────────────┼────────────────┤ │ ├───────────────────┼───────────────┼─────────────────────────────────────────────────────────────┤
│ http-proxy-middleware │ https://github.com/advisories/GHSA-c7qv-q95q-8v27 │ │ │ 1.3.1 │ 2.0.7, 3.0.3 │ http-proxy-middleware: Denial of Service │
│ │ │ │ │ │ │ https://avd.aquasec.com/nvd/cve-2024-21536 │
└───────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴─────────────────────────────────────────────────────────────┘
```

File: docker/test-images/zipkin-eureka/pom.xml [hunk 1]
```diff
@@ -29,20 +29,20 @@
       <dependency>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-dependencies</artifactId>
-        <version>3.3.4</version>
+        <version>3.3.5</version>
```
Comment (reta) [inline docker/test-images/zipkin-eureka/pom.xml]: I believe 3.3.6 is expected to land on Thu

Comment (reta) [reply docker/test-images/zipkin-eureka/pom.xml]: `3.3.6` is out!

**Comment by codefromthecrypt** [comment]
suggestion: why don't we bump, merge and release this (after updating deps unrelated to docker). In the release notes, we can say "help wanted" to fix the remaining CVEs in the UI. Some don't use the UI at all, and/or might not care about those CVEs. Meanwhile, if no one can help fix them, we shouldn't hold the rest of the project and downstream hostage.

@reta if cool by you, you can edit this branch about spring boot version and merge!

**Review by shakuzen — APPROVED** [review]
I'm not sure why the lint check is failing given https://github.com/openzipkin/zipkin/pull/3785/files#diff-23773f5014e0437c31c4c94b70d816279ceba1e308127698bda89bf9582b09d5R4-R5 but looks good to me

**Comment by reta** [comment]
> I'm not sure why the lint check is failing given https://github.com/openzipkin/zipkin/pull/3785/files#diff-23773f5014e0437c31c4c94b70d816279ceba1e308127698bda89bf9582b09d5R4-R5 but looks good to me

It is because this link `https://fluffy.cc/` is not available anymore, I will update docs shortly to remove it

**Review by shakuzen — APPROVED** [review]

**Comment by reta** [comment]
Merging, thanks @codefromthecrypt and @shakuzen !

**Pull request merged**